replica/table.cc

```diff
@@ -136,10 +139,6 @@ sstables::sstable_set table::make_maintenance_sstable_set() const {
             sstables::make_partitioned_sstable_set(_schema, use_level_metadata));
 }
 
-void table::refresh_compound_sstable_set() {
-    _sstables = make_compound_sstable_set();
-}
-
```

Comment:
if we want to avoid full refresh when a single sstable set changes, then compound set pointing to stable addresses is indeed a viable approach.

this approach is good in that compaction group no longer needs to be aware of compound set, and updates to each group are pulled automatically. 

if we think forward to tablets, the compound set for it will have to be aware of each compaciton_group, not individual sets as we want range scans to open a single group at a time, and filter out unwanted groups for single-key reads. See https://github.com/scylladb/scylladb/pull/13666/

table::make_compound_sstable_set() will have to be changed to produce one compound set for each group. so a tablet aware set will be managing a set of compound sets and their respective group ptrs (to collect metadata from).